server/routerlicious/packages/services-core/src/runWithRetry.ts

```diff
@@ -149,7 +149,27 @@ export async function runWithRetry<T>(
 export function shouldRetryNetworkError(error: any): boolean {
     if (error instanceof Error && error?.name === "NetworkError") {
         const networkError = error as NetworkError;
-        return !networkError.isFatal && networkError.canRetry !== undefined && networkError.canRetry;
+        return !networkError.isFatal && networkError.canRetry === true;
     }
     return false;
 }
+
+/**
+ * Helper function that calculates interval to wait before retrying. Leverage's {@link NetworkError.retryAfterMs}
+ * if the error is a {@link NetworkError}. Can be used with {@link runWithRetry} and {@link requestWithRetry}.
+ * @param {any} error - the error parameter to be inspected. If it is a {@link NetworkError},
+ * {@link NetworkError.retryAfterMs} will be used as the retry interval.
+ * @param {number} numRetries - the current retry count to be used in exponential backoff calculation.
+ * @param {number} retryAfterInterval - default value to be used when calculating the retry interval. Used when
+ * {@link NetworkError.retryAfterMs} is not defined.
+ */
+export function calculateRetryIntervalForNetworkError(
+    error: any,
+    numRetries: number,
+    retryAfterInterval: number): number {
+    let overwriteRetryAfterInterval: number | undefined;
+    if (error instanceof Error && error?.name === "NetworkError") {
+        overwriteRetryAfterInterval = (error as NetworkError).retryAfterMs;
+    }
+    return (overwriteRetryAfterInterval ?? retryAfterInterval) * 2 ** numRetries;
```

Comment:
If the error supplies us a retryAfter duration, I don't think we need to exponentially increase it. For example, if a 429 Throttled error gives us retryAfter 5s on our 2nd retry, we don't (shouldn't) need to wait 10s to retry. I think you can just return networkError.retryAfterMs if it is not undefined